.github/workflows/main.yml

```diff
@@ -46,22 +46,45 @@ jobs:
             git submodule update --init --recursive
             make html
 
-    # Deployment to GH Pages
-  deploy:
+  # Builds the WebPages locally with Publish settings before Deployment
+  publish:
     # Runner Type(s)
     runs-on: ubuntu-latest
-    # Dependents on Build 
-    needs: build
     strategy:
       matrix:
         python-version: [3.9]
 
+    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      - name: Displaying Python Versions
-        run: python -c "import sys; print(sys.version)"
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      # Set up Python Environment based on the version
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      # Installations for all the dependencies 
+      - name: Install dependencies
+        run: |
+            python -m pip install --upgrade pip
+            pip install -r requirements.in
+      # Add the support plugins required & builds the webpages in HTML format
+      - name: Build the webpages
+        run: |
+            git submodule update --init --recursive
+            make publish
```

Comment:
Is this `make publish` only to check that the site builds correctly? I wonder if we can see or at least store the resulting HTML artifacts somewhere.

Reply:
I think so this can be done but then how will we be able to check for the pages thus produced? ... need to check this. I will update on the same :)